api/v1beta1/azuremachine_types.go

```diff
@@ -122,6 +122,10 @@ type AzureMachineSpec struct {
 	// DNSServers adds a list of DNS Server IP addresses to the VM NICs.
 	// +optional
 	DNSServers []string `json:"dnsServers,omitempty"`
+
+	// VMExtensions specifies a list of extensions to be added to the virtual machine.
```

Comment:
Would be good to add a short doc on this feature to https://github.com/kubernetes-sigs/cluster-api-provider-azure/tree/main/docs/book/src/topics so it can show up in the book

Reply:
Just added in my latest revision. Let me know what you think!